exec/java-exec/src/main/java/org/apache/drill/exec/coord/zk/ZookeeperClient.java

```diff
@@ -301,6 +324,54 @@ public void put(final String path, final byte[] data, DataChangeVersion version)
     }
   }
 
+  public void putAsTransaction(Map<String, byte[]> pathsWithData) {
+    putAsTransaction(pathsWithData, null);
+  }
+
+  /**
+   * Puts the given sets of blob and their data's in a transactional manner. It expects all the blob path to exist
+   * before calling this api.
+   * @param pathsWithData - map of blob paths to update and the final data
+   * @param version - version holder
+   */
+  public void putAsTransaction(Map<String, byte[]> pathsWithData, DataChangeVersion version) {
```

Comment:
Do we currently use non null version. If not then can you please mention it in the comment that this is needed for future use.